pkg/registry/core/backupentry/strategy.go

```diff
@@ -81,8 +80,12 @@ func mustIncreaseGeneration(oldBackupEntry, newBackupEntry *core.BackupEntry) bo
 		return true
 	}
 
-	if kutil.HasMetaDataAnnotation(&newBackupEntry.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile) {
-		return true
+	if val, ok := newBackupEntry.ObjectMeta.Annotations[v1beta1constants.GardenerOperation]; ok {
+		if val == v1beta1constants.GardenerOperationReconcile ||
+			val == v1beta1constants.GardenerOperationMigrate ||
+			val == v1beta1constants.GardenerOperationRestore {
+			return true
+		}
```

Comment:
This will always increase the generation for every change as long as the above conditions are met. Instead, it should only be increased once when the annotation is set, correct?

Reply:
Yep, it would increase the generation for metadata changes (status updates won't do anything). 

However, I was worried about the following scenario:
1. Resource is annotated/updated and generation is increased
2. The resource is picked for reconciliation
3. The status is updated and the ObservedGeneration is set to equal the Generation
4. The controller crashes before being able to remove the operation annotation.

When the controller restarts it should requeue the BackupEntry via the `Add` event, but I was afraid if for some reason the `Add` event never gets fired. Then, when we actually want to reconcile the BackupEntry resource we won't be able to do that even by changing its Timestap annotation `metav1.SetMetaDataAnnotation(&backupEntry.ObjectMeta, v1beta1constants.GardenerTimestamp, TimeNow().UTC().String())` as this would not increase the the generation.
I'm really not sure if there's a possibility for `Add` events to be missed, as I haven't encountered such a case before, so I could improve the check to compare the old vs the new annotations, wdyt @rfranzke .

Reply:
Why should the resource be picked for reconciliation (step 2) after it was annotated with the "migrate" or "restore" operation? And even if, then there would be another attempt to set the same operation annotation by gardenlet, right?
So, what you write sounds a little artificial/spiritual to me 😄 I would vote to implement it "the proper way" and follow up if we see concrete issues with it. This current behaviour (and also what is happening in the `shoot/strategy.go`) might have undesired side effects.

Reply:
Spiritual describes it pretty nice :D. I'll adapt it to use the comparison of the old and new annotations.